logging.js

```diff
@@ -0,0 +1,168 @@
+var Command = require('./command');
+var debug = require('debug')('mozdevice:logging');
+
+// Cache the connection to logcat so we can re-use for additional MozDevices
+var currentProcess;
+var currentStream;
+
+/**
+ * API for interacting with a device's logging
+ * @param {Device} device
+ * @constructor
+ */
+var Logging = function(device) {
+  this.serial = device.serial;
+};
+
+/**
+ * Execute a command against `adb shell` contextual to the device serial
+ * @param command
+ * @returns {Promise}
+ */
+Logging.prototype.adbShell = function(command) {
+  return new Command()
+    .env('ANDROID_SERIAL', this.serial)
+    .adbShell(command)
+    .exec();
+};
+
+/**
+ * Write a message to the ADB log
+ * @param {string} priority log command priority, e.g. v, d, i, w, e
+ * @param {string} [tag] log command process tag, e.g. GeckoConsole, Homescreen
+ * @param {string} message message to write to the ADB log
+ * @returns {Promise}
+ */
+Logging.prototype.write = function(priority, tag, message) {
+  if (!message) {
+    message = tag;
+    tag = 'GeckoConsole';
+  }
+
+  debug('Writing to log');
+
+  var command = util.format('log -p %s -t %s "%s"', priority, tag, message);
+  return this.adbShell(command);
+};
+
+/**
+ * Write a verbose message to the ADB log
+ * @param {string} [tag] log command process tag, e.g. GeckoConsole, Homescreen
+ * @param {string} message message to write to the ADB log
+ * @returns {Promise}
+ */
+Logging.prototype.verbose = function(tag, message) {
+  return this.write('v', tag, message);
+};
+
+/**
+ * Write a debug message to the ADB log
+ * @param {string} [tag] log command process tag, e.g. GeckoConsole, Homescreen
+ * @param {string} message message to write to the ADB log
+ * @returns {Promise}
+ */
+Logging.prototype.debug = function(tag, message) {
+  return this.write('d', tag, message);
+};
+
+/**
+ * Write an info message to the ADB log
+ * @param {string} [tag] log command process tag, e.g. GeckoConsole, Homescreen
+ * @param {string} message message to write to the ADB log
+ * @returns {Promise}
+ */
+Logging.prototype.info = function(tag, message) {
+  return this.write('i', tag, message);
+};
+
+/**
+ * Write a warning message to the ADB log
+ * @param {string} [tag] log command process tag, e.g. GeckoConsole, Homescreen
+ * @param {string} message message to write to the ADB log
+ * @returns {Promise}
+ */
+Logging.prototype.warning = function(tag, message) {
+  return this.write('w', tag, message);
+};
+
+/**
+ * Write an error message to the ADB log
+ * @param {string} [tag] log command process tag, e.g. GeckoConsole, Homescreen
+ * @param {string} message message to write to the ADB log
+ * @returns {Promise}
+ */
+Logging.prototype.error = function(tag, message) {
+  return this.write('e', tag, message);
+};
+
+/**
+ * Clear the ADB log
+ * @returns {Promise}
+ */
+Logging.prototype.clear = function() {
+  debug('Clearing');
+  return this.adbShell('logcat -c');
```

Comment:
Actually, both of them work. I opted to use `adb shell logcat -c` to avoid having to create another prototype method for a single use case.